src/GitHub.UI.Reactive/Controls/ViewBase.cs

```diff
@@ -0,0 +1,223 @@
+                if (hasLoading != null)
+                {
+                    subs.Add(this.OneWayBind(hasLoading, x => x.IsLoading, x => x.IsLoading));
+                }
+
+                if (hasBusy != null)
+                {
+                    subs.Add(this.OneWayBind(hasBusy, x => x.IsBusy, x => x.IsBusy));
+                }
+
+                HasBusyState = hasLoading != null || hasBusy != null;
+                subscriptions = subs;
+            };
+
+            this.WhenActivated(d =>
+            {
+                d(this.Events()
+                    .KeyUp
+                    .Where(x => x.Key == Key.Escape && !x.Handled)
+                    .Subscribe(key =>
+                    {
+                        key.Handled = true;
+                        (this.ViewModel as IDialogViewModel)?.Cancel.Execute(null);
+                    }));
+            });
+        }
+
+        /// <summary>
+        /// The control finalizer.
+        /// </summary>
+        ~ViewBase()
+        {
+            Dispose(false);
+        }
+
+        /// <summary>
+        /// Gets or sets the control's data context as a typed view model.
+        /// </summary>
+        [AllowNull]
+        public TInterface ViewModel
+        {
+            [return: AllowNull]
+            get { return (TInterface)GetValue(ViewModelProperty); }
+            set { SetValue(ViewModelProperty, value); }
+        }
+
+        /// <summary>
+        /// Gets or sets the control's data context as a typed view model. Required for interaction
+        /// with ReactiveUI.
+        /// </summary>
+        [AllowNull]
+        TInterface IViewFor<TInterface>.ViewModel
+        {
+            [return: AllowNull]
+            get { return ViewModel; }
+            set { ViewModel = value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the control's data context. Required for interaction with ReactiveUI.
+        /// </summary>
+        [AllowNull]
+        object IViewFor.ViewModel
+        {
+            [return: AllowNull]
+            get { return ViewModel; }
+            set { ViewModel = (TInterface)value; }
+        }
+
+        /// <summary>
+        /// Gets or sets the control's data context. Required for interaction with ReactiveUI.
+        /// </summary>
+        [AllowNull]
+        IViewModel IView.ViewModel
+        {
+            [return: AllowNull]
+            get { return ViewModel; }
+        }
+
+        /// <summary>
+        /// Releases the managed resources held by the control.
+        /// </summary>
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Releases the managed or unmanaged resources held by the control.
+        /// </summary>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposing)
+            {
+                subscriptions?.Dispose();
+                subscriptions = null;
```

Comment:
This code should only be called once, and it is possible to call Dispose() multiple times, so it should be in a `if (!disposed)` block (I don't usually throw exceptions if Dispose is called multiple times because I figure it's a hard thing to control, and it's nice to be able to do it and not worry about whether some other code has already done it). Alternatively, you can do `if (disposed) return` right before this.